src/main/java/core/basesyntax/serviceimpl/ShopServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.operation.OperationHandler;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactionList) {
+        for (FruitTransaction transaction : transactionList) {
+            OperationHandler operationHandler =
+                    operationStrategy.get(transaction.getOperation());
+            operationHandler.operationType(transaction.getFruit(),
+                    transaction.getQuantity());
```

Comment:
To follow the SOLID principles, especially the Single Responsibility Principle, consider refactoring the 'operationType' method to have a more descriptive name that reflects the action it performs, and ensure that the OperationHandler classes are focused on a single task.